code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -219,6 +219,14 @@
 
 /obj/vehicle/sealed/mecha/attackby(obj/item/weapon, mob/living/user, list/modifiers)
 	if(user.combat_mode)
+		//If our weapon that we are hitting the mech with has armour penetration, we could potentially get a hit in on the occupant
+		var/peeling_the_onion = clamp(weapon.armour_penetration - (get_armor_rating(MELEE)/2), 0, 100)
+
+		if(peeling_the_onion && prob(peeling_the_onion) && !(mecha_flags & CANNOT_OVERPENETRATE) \
+			&& LAZYLEN(occupants) \
+			&& !(mecha_flags & SILICON_PILOT))
+			var/mob/living/hitmob = pick(occupants)
+			weapon.melee_attack_chain(user, hitmob, modifiers) //Perform an extra attack on the occupant if all the above conditions pass
```

Comment:
I think transferring 100% of the attack force (literally performing an entire attack) to the interior mob seems... wacky, like, if nothing else unrealistic 

The pilot should maybe take a portion of the damage

Reply:
I did this (with new technology)